open-banking-accelerator/components/com.wso2.openbanking.accelerator.common/src/main/java/com/wso2/openbanking/accelerator/common/util/JWTUtils.java

```diff
@@ -120,4 +137,120 @@ public static boolean validateJWTSignature(String jwtString, String jwksUri, Str
+     *
+     * @param signedJWT the signed JWT to be validated
+     * @param publicKey the public key that ois used for validation
+     * @param algorithm the algorithm expected to have signed the jwt
+     * @return true if signature is valid else false
+     * @throws NoSuchAlgorithmException if the given algorithm doesn't exist
+     * @throws InvalidKeySpecException if the provided key is invalid
+     * @throws JOSEException if an error occurs during the signature validation process
+     */
+    public static boolean validateJWTSignature(SignedJWT signedJWT, String publicKey, String algorithm) throws
+            NoSuchAlgorithmException, InvalidKeySpecException, JOSEException {
+
+        byte[] publicKeyData = Base64.getDecoder().decode(publicKey);
+        X509EncodedKeySpec spec = new X509EncodedKeySpec(publicKeyData);
+        KeyFactory kf = KeyFactory.getInstance(algorithm);
+        RSAPublicKey rsapublicKey = (RSAPublicKey) kf.generatePublic(spec);
+        JWSVerifier verifier = new RSASSAVerifier(rsapublicKey);
+        return signedJWT.verify(verifier);
+
+    }
+
+    /**
+     * Validate legitimacy of JWT.
+     *
+     * @param jwtString JWT string
+     */
+    public static boolean isJWT(String jwtString) {
+
+        if (jwtString == null) {
+            return false;
+        }
+        if (StringUtils.isBlank(jwtString)) {
+            return false;
+        }
+        if (StringUtils.countMatches(jwtString, DOT_SEPARATOR) != 2) {
+            return false;
+        }
+        try {
+            //Checking whether the jwtString is jwt parsable.
+            JWTParser.parse(jwtString);
+            return true;
+        } catch (ParseException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Provided JWT is not a parsable.", e);
+            }
+            return false;
+        }
+
+    }
+
+    /**
+     * Parses the provided JWT string into a SignedJWT object.
+     *
+     * @param jwtString the JWT string to parse
+     * @return the parsed SignedJWT object
+     * @throws IllegalArgumentException if the provided token identifier is not a parsable JWT
+     * Will not throw ParseException as it is already validated by isJWT
+     */
+    public static SignedJWT getSignedJWT(String jwtString) throws ParseException {
+
+        if (isJWT(jwtString)) {
+            return SignedJWT.parse(jwtString);
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("Provided token identifier is not a parsable JWT.");
+            }
+            throw new IllegalArgumentException("Provided token identifier is not a parsable JWT.");
+        }
+
+    }
+
+    /**
+     * Validates whether a given JWT is not expired.
+     *
+     * @param signedJWT  the signed jwt that needs to validated
+     * @return true if the jwt is not expired
+     */
+    public static boolean validateExpiryTime(SignedJWT signedJWT) throws ParseException {
```

Comment:
Check if timezone can affect this logic

Reply:
Here I am using logic directly implemented in IS @PushJWTValidator but can't use it as it's private